test/functional/test-dom.js

```diff
@@ -465,6 +465,7 @@ describe('DOM', () => {
       const sibling = document.createElement('div');
       expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
       parent.appendChild(element);
+      expect(dom.hasNextNodeInDocumentOrder(element)).to.be.false;
```

Comment:
No need for grandparent. We only need one parentNode (that neither the element or the parent satisfies the nextSibling condition).

Happy to add some more if you feel this is not enough.